### ISSUE TITLE
Second increment crushes on Mongoose 3.1.2.

### ISSUE THREAD

**Issue by lbeschastny** [mongoosejs/mongoose-number#1]
The following code

``` coffee-script
mongoose = require 'mongoose'
require("mongoose-number") mongoose
async = require 'async'

mongoose.connect "mongodb://localhost:27017/test"

Schema = mongoose.Schema
ObjectId = Schema.ObjectId

SomeSchema = new Schema
  counter: type: 'MongooseNumber', default: 0

mongoose.model 'Some', SomeSchema
Some = mongoose.model 'Some'

test = (some) ->
  console.log 'first'
  some.counter.increment()
  console.log 'second'
  try
    some.counter.increment()
  catch err
    console.log 'error!'
    console.log err.stack
  console.log 'saving...'
  some.save (err) -> console.log err ? 'done!'

run = ->
  some = new Some
  some.save ->
    test some

run()
```

crushes on the second `increment`. Console output:

```
first
second
error!
TypeError: Cannot read property 'schema' of undefined
    at Number.<anonymous> (/home/leonid/Documents/intervox/avocado/avocado/node_modules/mongoose-number/lib/type.js:36:30)
    at /home/leonid/Documents/intervox/avocado/avocado/test.coffee:34:20
    at Object.<anonymous> (/home/leonid/Documents/intervox/avocado/avocado/test.coffee:45:3)
    at Object.<anonymous> (/home/leonid/Documents/intervox/avocado/avocado/test.coffee:47:4)
    at Module._compile (module.js:446:26)
    at Object.run (/home/leonid/local/lib/node_modules/coffee-script/lib/coffee-script/coffee-script.js:68:25)
    at /home/leonid/local/lib/node_modules/coffee-script/lib/coffee-script/command.js:135:29
    at /home/leonid/local/lib/node_modules/coffee-script/lib/coffee-script/command.js:110:18
    at [object Object].<anonymous> (fs.js:123:5)
    at [object Object].emit (events.js:64:17)
saving...
done!
```

**Comment by lbeschastny** [comment]
It seems like `some.update $inc: counter: 1` works great.

**Comment by aheckmann** [comment]
patching soon

**Comment by aheckmann** [comment]
fixed in 0.0.2
